package.json

```diff
@@ -4,12 +4,12 @@
   "scripts": {
     "test:unit": "jest",
     "test:lint": "eslint .",
-    "test:types": "tsc",
+    "test:types": "tsc --build packages/*",
     "pretest": "npm run build:bundle",
     "test": "npm run test:unit && npm run test:lint && npm run test:types",
     "build:bundle": "rollup -c",
-    "prebuild:types": "rimraf \"packages/**/*.d.ts\"",
-    "build:types": "tsc -p tsconfig.declarations.json",
+    "prebuild:types": "rimraf \"packages/**/*.+(d.ts|.tsbuildinfo)\"",
```

Comment:
By the way, if I recall correctly, this was the way I had addressed the same sort of issue you encountered at https://github.com/WordPress/gutenberg/pull/18942#issuecomment-595991723 . It's a bit ugly to need to remove the files, but it works.

I wonder though, does it not defeat the purpose of `.tsbuildinfo` to remove it before building types? Isn't that file meant to be some sort of cache to improve performance of subsequent builds?

Reply:
As far as I've been able to understand, the `.tsbuildinfo` files serve as a cache and a mechanism to tell TypeScript when it doesn't need to generate any files. They will help save work and speed up builds.

I believe it's best to leave them alone as much as possible. Before publishing a package it makes sense to clean in case declaration files that should not appear in the published package have been generated. I've discovered `tsc --build --clean` which seems to be the best way to handle this.

With the exception of before publishing, it's best to leave these files alone and let `tsc` manage them.